components/dashboard/src/data/workspaces/stop-workspace-mutation.ts

```diff
@@ -8,20 +8,26 @@ import { useMutation } from "@tanstack/react-query";
 import { workspacesService } from "../../service/public-api";
 import { getGitpodService } from "../../service/service";
 import { useFeatureFlag } from "../featureflag-query";
+import { useToast } from "../../components/toasts/Toasts";
 
 type StopWorkspaceArgs = {
     workspaceId: string;
 };
 
 export const useStopWorkspaceMutation = () => {
     const usePublicApiWorkspacesService = useFeatureFlag("publicApiExperimentalWorkspaceService");
+    const { toast } = useToast();
 
     // No need to manually update workspace in cache here, we'll receive messages over the ws that will update it
     return useMutation({
         mutationFn: async ({ workspaceId }: StopWorkspaceArgs) => {
-            return usePublicApiWorkspacesService
-                ? workspacesService.stopWorkspace({ workspaceId })
-                : getGitpodService().server.stopWorkspace(workspaceId);
+            return (
+                usePublicApiWorkspacesService
+                    ? workspacesService.stopWorkspace({ workspaceId })
+                    : getGitpodService().server.stopWorkspace(workspaceId)
+            ).catch((err) => {
```

Comment:
Here, I think using `try/catch` would be more idiomatic of the `async/await` pattern. Alternatively, it may more appropriate if the caller actually chooses to show the error message, as in this mutation we don't really know the context, and it doesn't give the caller a chance to decide how to handle it - retry, swallow or otherwise

Reply:
Updated use try/catch and move it to the caller, tested that catching the `.mutate` function still works as well, the error is still shown